src/util.rs

```diff
@@ -0,0 +1,38 @@
+/// HACK! To reduce code duplication in the generated binary, this macro is used to generate a
```

Comment:
Considering this is only used in two places in the entire codebase is the complexity of this macro worth it? I feel like this macro makes the code quite a lot harder to understand, for what I don't see as a big gain in terms of maintainability

Reply:
Eh, whilst I agree with you, we already know that formatting machinery is particularly bloatful in Rust, and this hack has significant impact in reducing it in Veil - imagine all the stuff in private.rs, but duplicated once more for generating a String rather than using a std::format::Formatter. Bleh.

One alternative; as you mention it's only used in two places throughout the binary. I suppose we could get rid of the macro and just DIY the formatter hack at the call sites? That would be a lot easier to maintain & understand.

Reply:
I think I like the second one more, but then I'm worried about us using the formatting hack in other places and then regretting we didn't use the macro. 
I don't know, I'll leave the decision up to you

Reply:
Whilst I love the macro like its my own child the thing is way too complex to justify its existence. I shall go with the second one, im sure we can dig it out of the git abyss if ever needed again, but I kinda doubt that it will be anyways

Reply:
Actually wouldn't something like:
```
struct GiveMeAFormatter<F: Fn(&mut fmt::Formatter) -> fmt::Result>(F);
impl<F> fmt::Display for GiveMeAFormatter<F>
where
    for <'r, 's> F: Fn(&'r mut std::fmt::Formatter<'s>) -> fmt::Result,
{
    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
        self.0(f)
    }
}

pub fn give_me_a_formatter<F: Fn(&mut std::fmt::Formatter) -> fmt::Result>(f: F) -> String {
    format!("{}", GiveMeAFormatter(f))
}
```

also work?

Reply:
lmfao 💀 not sure why I reached for the macro

Reply:
when you spend the entire day using a hammer everything starts to look like a nail